pkg/cmd/docs-issue-generation/github.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+func searchCockroachPRs(startTime time.Time, endTime time.Time) ([]cockroachPR, error) {
+	prCommitsToExclude, err := searchJiraDocsIssues(startTime)
+	if err != nil {
+		fmt.Println(err)
+	}
+	hasNextPage, nextCursor, prs, err := searchCockroachPRsSingle(startTime, endTime, "", prCommitsToExclude)
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+	result := prs
+	for hasNextPage {
+		hasNextPage, nextCursor, prs, err = searchCockroachPRsSingle(startTime, endTime, nextCursor, prCommitsToExclude)
+		if err != nil {
+			fmt.Println(err)
+			return nil, err
+		}
+		result = append(result, prs...)
+	}
+	return result, nil
+}
+
+func searchCockroachPRsSingle(
+	startTime time.Time,
+	endTime time.Time,
+	cursor string,
+	prCommitsToExclude map[int]map[string]string,
+) (bool, string, []cockroachPR, error) {
+	var search gqlCockroachPR
+	var result []cockroachPR
+	query := `query ($cursor: String, $ghSearchQuery: String!) {
+	  search(first: 100, query: $ghSearchQuery, type: ISSUE, after: $cursor) {
+	    nodes {
+	      ... on PullRequest {
+	        title
+	        number
+	        body
+	        baseRefName
+	        commits(first: 100) {
+	          edges {
+	            node {
+	              commit {
+	                oid
+	                messageHeadline
+	                messageBody
+	              }
+	            }
+	          }
+	          pageInfo {
+	            hasNextPage
+	            endCursor
+	          }
+	        }
+	      }
+	    }
+	    pageInfo {
+	      hasNextPage
+	      endCursor
+	    }
+	  }
+	}`
+	queryVariables := map[string]interface{}{
+		"ghSearchQuery": fmt.Sprintf(
+			`repo:cockroachdb/cockroach is:pr is:merged merged:%s..%s`,
+			startTime.Format(time.RFC3339),
+			endTime.Format(time.RFC3339),
+		),
+	}
+	if cursor != "" {
+		queryVariables["cursor"] = cursor
+	}
+	err := queryGraphQL(query, queryVariables, &search)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Nit: since we're returning the error anyway, there's no reason to print it here. Someone else can handle it.